Utils/Example/CMakeLists.txt

```diff
@@ -0,0 +1,65 @@
+#  SPDX-License-Identifier: Apache-2.0
+#  ----------------------------------------------------------------------------
+#  Copyright 2021 Arm Limited
+#
+#  Licensed under the Apache License, Version 2.0 (the "License"); you may not
+#  use this file except in compliance with the License. You may obtain a copy
+#  of the License at:
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#  License for the specific language governing permissions and limitations
+#  under the License.
+#  ----------------------------------------------------------------------------
+
+# CMake configuration
+cmake_minimum_required(VERSION 3.15)
+include(ExternalProject)
+
+project(astcencoder_example VERSION 1.0.0)
+
+# Command line configuration
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
+set(CMAKE_EXPORT_COMPILE_COMMANDS 1)
+
+# Add the external project and pull out the project directories we need
+
+# Add e.g. "CMAKE_CACHE_ARGS -DISA_SSE41:String=ON" to ExternalProject_Add to
+# set configure settings usually found on the astcenc configure command line.
+# Remember that changing ISA changes the name of the library binary produced.
+
+ExternalProject_Add(astcencoder
```

Comment:
I like `ExternalProject` command but I haven't seen much use of this in cmake projects. Usually an `add_subdirectory` is the most common way of using TPIP. With external project every time you want to do a clean rebuild, you nuke the whole download and start again. With subdirectory, the dependency needs to be a "sub-directory" for that to work but all of the following explicit methods like `target_include_directories/target_link_directories` goes away.

Reply:
Might add that as a separate example - I can see both being useful to have.